src/drivers/opineo/driver.rs

```diff
@@ -197,15 +221,38 @@ impl Driver {
         })
     }
 
-    fn start_tap(&mut self) -> Event {
+    fn start_click(&mut self) -> Vec<Event> {
         log::trace!("Started CLICK event.");
-        self.is_tapped = true;
-        Event::TouchButton(TouchButtonEvent::Left(BinaryInput { pressed: true }))
+        log::trace!("First touch elapsed: {:?}", self.first_touch.elapsed());
+        log::trace!("Last touch elapsed: {:?}", self.last_touch.elapsed());
+        self.is_clicked = true;
+        let mut events = Vec::new();
+
+        let event = Event::TouchButton(TouchButtonEvent::Left(BinaryInput { pressed: true }));
+        events.push(event);
+        // The touchpad doesn't have a force sensor. The deck target wont produce a "click"
+        // event in desktop or lizard mode without a force value. Simulate a 1/4 press to work
+        // around this.
+        let event = Event::Trigger(TriggerEvent::PadForce(TriggerInput {
+            value: PAD_FORCE_NORMAL,
+        }));
+        events.push(event);
+        events
     }
 
-    fn release_tap(&mut self) -> Event {
+    fn release_click(&mut self) -> Vec<Event> {
         log::trace!("Released CLICK event.");
-        self.is_tapped = false;
-        Event::TouchButton(TouchButtonEvent::Left(BinaryInput { pressed: false }))
+        log::trace!("First touch elapsed: {:?}", self.first_touch.elapsed());
+        log::trace!("Last touch elapsed: {:?}", self.last_touch.elapsed());
+        self.is_clicked = false;
+        let mut events = Vec::new();
+        let event = Event::TouchButton(TouchButtonEvent::Left(BinaryInput { pressed: false }));
+        events.push(event);
+        // The touchpad doesn't have a force sensor. The deck target wont produce a "click"
+        // event in desktop or lizard mode without a force value. Simulate a 1/4 press to work
+        // around this.
+        let event = Event::Trigger(TriggerEvent::PadForce(TriggerInput { value: 0 }));
+        events.push(event);
```

Comment:
Instead of emulating the touch force in the source device, can we instead handle this in the target device(s)?